### PR TITLE
Feat/add first getter and setter methods

### PR DESCRIPTION
**Pull request by MrBalum**
This PR will add necessary setter and getter to the Barcode Wrapper, Those methods will help to define which barcodes types we should want to read.  As Example, you can only read QR Codes if you set reading qr codes on true

Please use feature/testsuite branch to test the new changes

### CODE REVIEW

File: src/Traits/SoftekGetterTrait.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?php
+declare(strict_types=1);
+
+namespace DexproSolutionsGmbh\SoftekBarcodeWrapper\Traits;
+
+use FFI\CData;
+
+/**
+ * This Trait can be implemented in each business logic class like transaction classes, service classes
+ * or controller classes. Be sure that the global properties instance and ffi are declared and usable.
```
Comment (FabianTe) [inline src/Traits/SoftekGetterTrait.php:10]: I think we should talk about the terms `business logic`, `transaction classes` and `service classes`.

This package implements a wrapper arround a FFI. That is a rather technical topic. In this context I really don't know what a transaction class or service class is supposed to be. Of course, those terms are often used when implementing a web app / api. But this package has nothing to do with that. IMO `SoftekBarcodeAPI` isn't a service class or anything similar. In the context of this package, these terms really have no meaning to me.

I think, here less is more: "This trait implements getter / setter methods on the ffi object wrapped by this library."

Also, traits are not *implemented* by classes, they are *used*.

File: src/Traits/SoftekGetterTrait.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * Be sure that {@link SoftekSetterTrait::setLicenseKey()} is called before any getter is used.
+ * Notice: Most times getter functions within the barcode tool kit requires a premature function call of mtScanBarCode
+ */
+trait SoftekGetterTrait
```
Comment (FabianTe) [inline src/Traits/SoftekGetterTrait.php:14]: Are these traits ment for public use? Or are they internal to this package? If the latter, please mark these classes with `@internal`.

File: src/SoftekBarcodeAPI.php [hunk 3]
```diff
@@ -100,9 +106,12 @@ private function extractScanResults(int $resultCount): array
...
             $right = FFI::new($ffiType);
 
-            $this->ffi->mtGetBarStringPos($this->instance, $barcodeIndex, FFI::addr($left), FFI::addr($top), FFI::addr($right), FFI::addr($bottom));
+            // we pass pointer that we create with the FFI::addr, please ignore the pointer passed here at
+            // this point it is not comparable to php
+            $this->getBarcodePosition($barcodeIndex, FFI::addr($left), FFI::addr($top), FFI::addr($right), FFI::addr($bottom));
 
-            // set referenced types
+            // set referenced types and access on the position variables, the variables are declared
+            // within the barcode tool kit library
```
Comment (FabianTe) [inline src/SoftekBarcodeAPI.php:114]: You will have to explain these comments to me. I can't understand what they mean.

**Review by FabianTe — CHANGES_REQUESTED** [review]
My comments relate mostly to the documentation / code comments. Let's talk about those in person, resolve them and then merge since otherwise the changes are sensible :+1:

File: src/Traits/SoftekGetterTrait.php (hunk 1 above)
Comment (MrBalum) [reply src/Traits/SoftekGetterTrait.php]: I have adjusted the comments

**Pull request merged**